src/components/display/components/StickyToggleButtonComponent.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useState } from 'react'
+import { StickyToggleButtonArgs } from 'display-api'
+import { ButtonComponent } from './ButtonComponent'
+import { setMethodVariable } from '../method'
+
+export const StickyToggleButtonComponent = ({ button, name, on_change, default_value }: StickyToggleButtonArgs) => {
+	const [isEnabled, setEnabled] = useState<boolean>(default_value)
+
+	const onChange = () => {
+		setEnabled(!isEnabled)
+		name && setMethodVariable(name, isEnabled)
+	}
+
+	return (
+		<ButtonComponent
+			text={button.text}
+			color_scheme={isEnabled ? button.color_scheme : 'gray'}
+			variant={button.variant}
+			icon={button.icon}
+			size={button.size}
+			disabled={button.disabled}
+			readonly={button.readonly}
+			interact={on_change}
```

Comment:
To avoid confusion, I think it makes a little more sense to trigger the `on_change` method in the onChange function above, especially to guarantee that the variable will be set before calling the function.

Reply:
Makes sense. I assigned it to the interact of the button component to avoid redundant implementation since the ActionObject has multiple checks and functions to execute.

I think I'll move the ActionObject implementation to a more global space and then will do what you're suggesting. I'll update this PR.

Reply:
My suggestion above is requiring a couple of changes so for now I have implemented the on_change in the stickyToggleButtonComponent itself. I'll refactor the logic to move all of the actionObject handling to one place when we have some bandwidth.